internal/kafka/internal/workers/cluster_mgrs/cleanup_clusters_mgr.go

```diff
@@ -95,6 +101,20 @@ func (m *CleanupClustersManager) processCleanupClusters() error {
 	return errList
 }
 
+func (m *CleanupClustersManager) reconcileCleanupEnterpriseCluster(cluster api.Cluster) error {
+	glog.Infof("Removing Enterprise Dataplane cluster %s fleetshard service account", cluster.ClusterID)
+	serviceAccountRemovalErr := m.kasFleetshardOperatorAddon.RemoveServiceAccount(cluster)
+	if serviceAccountRemovalErr != nil {
+		return errors.Wrapf(serviceAccountRemovalErr, "failed to removed Dataplane cluster %s fleetshard service account", cluster.ClusterID)
+	}
+	glog.Infof("Soft deleting the Enterprise Dataplane cluster %s from the database", cluster.ClusterID)
+	deleteError := m.clusterService.DeleteByClusterID(cluster.ClusterID)
```

Comment:
This method only soft delete the cluster - see https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/c6de1499fafb2d6c86b3d1d93c25404e78addb8f/internal/kafka/internal/services/clusters.go#L393

To allow for a flow like:
- register cluster
- deregister cluster
- register cluster

We might need to hard delete the record from the database.

We'll need to use `db.Unscoped()` to perform a hard delete of enterprise clusters - see https://gorm.io/docs/delete.html#Delete-permanently for more.

Reply:
Agree, we might want to hard delete it to support the case where a previously de-registered cluster is registered again. Those clusters are long living, so this could happen.

Reply:
In that case wouldn't a different `ID` be created?